client/my-sites/plans/ecommerce-trial/index.tsx

```diff
@@ -0,0 +1,389 @@
+import { Button, Card } from '@automattic/components';
+import { useLocale } from '@automattic/i18n-utils';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import SegmentedControl from 'calypso/components/segmented-control';
+import BodySectionCssClass from 'calypso/layout/body-section-css-class';
+import {
+	getCurrentPlan,
+	getECommerceTrialDaysLeft,
+	getECommerceTrialExpiration,
+	isECommerceTrialExpired,
+} from 'calypso/state/sites/plans/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import DoughnutChart from '../doughnut-chart';
+import TrialFeatureCard from './trial-feature-card';
+
+import './style.scss';
+
+const ECommerceTrialPlans = () => {
+	const selectedSiteId = useSelector( ( state ) => getSelectedSiteId( state ) ) || -1;
+
+	const { currentPlan, eCommerceTrialDaysLeft, isTrialExpired, eCommerceTrialExpiration } =
+		useSelector( ( state ) => ( {
+			currentPlan: getCurrentPlan( state, selectedSiteId ),
+			isTrialExpired: isECommerceTrialExpired( state, selectedSiteId ),
+			eCommerceTrialDaysLeft: Math.round( getECommerceTrialDaysLeft( state, selectedSiteId ) || 0 ),
+			eCommerceTrialExpiration: getECommerceTrialExpiration( state, selectedSiteId ),
+		} ) );
+
+	const locale = useLocale();
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const trialStart = moment( currentPlan?.subscribedDate );
+	const trialEnd = moment( currentPlan?.expiryDate );
+	const trialDuration = trialEnd.diff( trialStart, 'days' );
+
+	/**
+	 * Trial progress from 0 to 1
+	 */
+	const trialProgress = 1 - eCommerceTrialDaysLeft / trialDuration;
+	const eCommerceTrialDaysLeftToDisplay = isTrialExpired ? 0 : eCommerceTrialDaysLeft;
+
+	// moment.js doesn't have a format option to display the long form in a localized way without the year
+	// https://github.com/moment/moment/issues/3341
+	const readableExpirationDate = eCommerceTrialExpiration?.toDate().toLocaleDateString( locale, {
+		month: 'long',
+		day: 'numeric',
+	} );
+
+	const [ interval, setInterval ] = useState( 'yearly' );
```

Comment:
Two thoughts here:
1. The default interval should be based on the interval for the user's current plan
2. It may be worth using `useState<PlanInterval>( 'yearly' );` here so we can apply type checks to the downstream `setInterval()` calls.

Reply:
Revolving conversation since we are removing the interval selection from this PR.